pkg/reconciler/channel/channel.go

```diff
@@ -96,9 +112,97 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, c *v1.Channel) pkgreconc
+	applyingEventPoliciesForBackingChannel, err := auth.GetEventPoliciesForResource(r.eventPolicyLister, backingChannel.GroupVersionKind(), backingChannel.ObjectMeta)
+	if err != nil {
+		return fmt.Errorf("could not get applying EventPolicies for for backing channel %s/%s: %w", channel.Namespace, channel.Name, err)
+	}
+
+	selector, err := labels.ValidatedSelectorFromSet(resources.LabelsForBackingChannelsEventPolicy(backingChannel))
+	if err != nil {
+		return fmt.Errorf("could not get valid selector for backing channels EventPolicy %s/%s: %w", backingChannel.Namespace, backingChannel.Name, err)
+	}
+
+	existingEventPoliciesForBackingChannel, err := r.eventPolicyLister.EventPolicies(backingChannel.Namespace).List(selector)
+	if err != nil {
+		return fmt.Errorf("could not get existing EventPolicies in backing channels namespace %q: %w", backingChannel.Namespace, err)
+	}
+
+	for _, policy := range existingEventPoliciesForBackingChannel {
+		if !r.containsPolicy(policy.Name, applyingEventPoliciesForBackingChannel) {
+
+			// the existing policy is not in the list of applying policies anymore --> is outdated --> delete it
+			err := r.eventingClientSet.EventingV1alpha1().EventPolicies(policy.Namespace).Delete(ctx, policy.Name, metav1.DeleteOptions{})
+			if err != nil && apierrs.IsNotFound(err) {
+				return fmt.Errorf("could not delete old EventPolicy %s/%s: %w", policy.Namespace, policy.Name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *Reconciler) reconcileBackingChannelEventPolicy(ctx context.Context, backingChannel *eventingduckv1.Channelable, eventpolicy *eventingv1alpha1.EventPolicy) error {
+	expected := resources.MakeEventPolicyForBackingChannel(backingChannel, eventpolicy)
+
+	foundEP, err := r.eventPolicyLister.EventPolicies(expected.Namespace).Get(expected.Name)
+	if apierrs.IsNotFound(err) {
+		_, err := r.eventingClientSet.EventingV1alpha1().EventPolicies(expected.Namespace).Create(ctx, expected, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("could not create EventPolicy %s/%s: %w", expected.Namespace, expected.Name, err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("could not get EventPolicy %s/%s: %w", expected.Namespace, expected.Name, err)
+	} else if r.policyNeedsUpdate(foundEP, expected) {
+		expected.SetResourceVersion(foundEP.ResourceVersion)
+		_, err := r.eventingClientSet.EventingV1alpha1().EventPolicies(expected.Namespace).Update(ctx, expected, metav1.UpdateOptions{})
+		if err != nil {
+			return fmt.Errorf("could not update EventPolicy %s/%s: %w", expected.Namespace, expected.Name, err)
+		}
+	}
+
 	return nil
 }
 
+func (r *Reconciler) containsPolicy(name string, policies []*eventingv1alpha1.EventPolicy) bool {
+	for _, policy := range policies {
+		if policy.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *Reconciler) policyNeedsUpdate(foundEP, expected *eventingv1alpha1.EventPolicy) bool {
+	return !equality.Semantic.DeepDerivative(expected, foundEP)
```

Comment:
should we just compare the spec and labels? Having a stale status is fine and we don't want to re-trigger an update for that, right?

Reply:
I thought about limiting this too. But then checked on the `DeepDerivative()`, and this ignores empty fields in `expected`. So we don't have to update this, when we update the `MakeEventPolicyForBackingChannel()` and set another field in `.metadata` :shrug:

Reply:
Does this make sense?